cmd_list.go

```diff
@@ -165,6 +166,148 @@ func (m *Miniredis) cmdLindex(c *server.Peer, cmd string, args []string) {
 	})
 }
 
+// LPOS key element [RANK rank] [COUNT num-matches] [MAXLEN len]
+func (m *Miniredis) cmdLpos(c *server.Peer, cmd string, args []string) {
+	if len(args) == 1 {
+		setDirty(c)
+		c.WriteError(errWrongNumber(cmd))
+		return
+	}
+	if !m.handleAuth(c) {
+		return
+	}
+	if m.checkPubsub(c, cmd) {
+		return
+	}
+
+	withTx(m, c, func(c *server.Peer, ctx *connCtx) {
+		db := m.db(ctx.selectedDB)
+		key, element := args[0], args[1]
+		t, ok := db.keys[key]
+		if !ok {
+			// No such key
+			c.WriteNull()
+			return
+		}
+		if t != "list" {
+			c.WriteError(msgWrongType)
+			return
+		}
+		l := db.listKeys[key]
+
+		// Default options.
+		rank := 1
+		count := 1
+		maxlen := len(l)
+
+		// Extract options from arguments if present.
+		//
+		// Redis allows duplicate options and uses the last specified.
+		// `LPOS key term RANK 1 RANK 2` is effectively the same as
+		// `LPOS key term RANK 2`
+		if len(args)%2 == 1 {
+			c.WriteError(msgSyntaxError)
+			return
+		}
+		var countSpecified bool
+		if len(args) > 2 {
+			for i := 2; i < len(args); i++ {
+				if i%2 == 0 {
+					val := args[i+1]
+					var err error
+					switch strings.ToLower(args[i]) {
+					case "rank":
+						if rank, err = strconv.Atoi(val); err != nil {
+							c.WriteError(msgInvalidInt)
```

Comment:
All these `c.WriteError()`s need to call `setDirty(c)` (for transactions) before returning.

Reply:
Done.